libs/core/langchain_core/runnables/base.py

```diff
@@ -753,7 +753,7 @@ def with_config(
         config: Optional[RunnableConfig] = None,
         # Sadly Unpack is not well supported by mypy so this will have to be untyped
         **kwargs: Any,
-    ) -> Runnable[Input, Output]:
+    ) -> RunnableBinding[Input, Output]:
```

Comment:
why?

Reply:
was causing typing issues (if somethings RunnableSerializable and then you update with `with_config` it was becoming just Runnable)

Reply:
also isn't it best practice to type outputs as specifically as possible

Reply:
then let's make it return RunnableSerializable. The idea with returning a looser type is to allow subclasses to override it (eg because they handle config themselves